glide.yaml

```diff
@@ -27,7 +27,7 @@ import:
 - package: github.com/go-ini/ini
   version: ^1.21.0
 - package: github.com/projectcalico/libcalico-go
-  version: v1.7.3
+  version: v1.7.x-series
```

Comment:
Is it definitely best practice now to have a version here that is not a fixed point?  (It may be - I'm just not confident that I've heard the latest thinking on this.  IIRC,
- when we were first using glide we typically had `version: master` in most places
- then I think we determined that that was wrong / dangerous, and generally shifted to using fixed SHA-1s.

I'm not sure where we are on that question now.)

Reply:
I think a released tag is best, e.g. v1.7.4; but I don't think we're ready yet to cut a libcalico-go release, so I thought pinning to the branch would make the intent clear.

But I could go either way - if you prefer a hash, I'll update the PR.

Reply:
Oh, btw, master is pinned to master in typha, if that makes your decision any easier :-)

Reply:
I'm happy with this myself.  I have a worry that @fasaxc had a concern about it - but we can always come back to that if so...

Reply:
@neiljerram - yeah, I think you're right about @fasaxc not caring for pinning to a branch. Seems fine to go back and forth depending on the preference of whomever is stuck with the pleasant task of rev'ing libcalico-go :-)

Reply:
I prefer tags or hashes because they're exact.  If we let it float on a branch then the behaviour depends on glide's cache and mirroring settings, which is dangerous.  We've ended up reverting fixes that way:

- one of us updates libcalico-go to branch-x == commit-y
- another user just runs "make update-vendor" (to pick up a fix to another dependency, say) but their cache or mirror is stale so for them branch-x is *older* than commit-y and they undo the first developer's work

It's subtle and hard to spot, hence the preference for pinning to commits/tags.

The auto-rev script does pin to master, which is something we've been meaning to fix.